buildSrc/src/main/java/quilt/internal/tasks/jarmapping/MapNamedJarTask.java

```diff
@@ -2,59 +2,27 @@
 
 import java.util.Map;
 
-import org.gradle.api.file.RegularFileProperty;
-import org.gradle.api.tasks.InputFile;
-import org.gradle.api.tasks.TaskAction;
 import quilt.internal.Constants;
-import quilt.internal.tasks.DefaultMappingsTask;
-import quilt.internal.tasks.build.MergeTinyV2Task;
+import quilt.internal.tasks.build.AddProposedMappingsTask;
 import quilt.internal.tasks.setup.DownloadMinecraftLibrariesTask;
-import quilt.internal.util.JarRemapper;
 
-public class MapNamedJarTask extends DefaultMappingsTask {
+public class MapNamedJarTask extends MapJarTask {
     public static final String TASK_NAME = "mapNamedJar";
 
-    @InputFile
-    private final RegularFileProperty jar;
-
-    @InputFile
-    private final RegularFileProperty mappings;
-
     public MapNamedJarTask() {
-        super(Constants.Groups.MAP_JAR_GROUP);
+        super(Constants.Groups.MAP_JAR_GROUP, Constants.PER_VERSION_MAPPINGS_NAME, "named");
 
         getInputs().files(getTaskByName(DownloadMinecraftLibrariesTask.TASK_NAME).getOutputs().getFiles().getFiles());
-        getOutputs().file(this.fileConstants.namedJar);
-
-        jar = getProject().getObjects().fileProperty();
-        mappings = getProject().getObjects().fileProperty();
 
-
-        jar.convention(() -> fileConstants.unpickedJar);
-        mappings.convention(getTaskByType(MergeTinyV2Task.class)::getOutputMappings);
+        inputJar.convention(() -> fileConstants.unpickedJar);
+        mappingsFile.convention(() -> this.<AddProposedMappingsTask>getTaskByName("insertAutoGeneratedMappings").getOutputMappings());
+        outputJar.convention(() -> fileConstants.namedJar);
 
         this.dependsOn("mergeTinyV2", "unpickHashedJar");
```

Comment:
It looks like the task now should depend on `insertAutoGeneratedMappings` instead of `mergeTinyV2`

Reply:
```suggestion
        this.dependsOn("insertAutoGeneratedMappings", "unpickHashedJar");
```